lib/ecto/adapters/postgres/connection.ex

```diff
@@ -1094,12 +1094,45 @@ if Code.ensure_loaded?(Postgrex) do
       end
     end
 
+    defp column_type(:identity, opts) do
+      start_value = [Keyword.get(opts, :start_value)]
+      increment   = [Keyword.get(opts, :increment)]
+      type_name   = ecto_to_db(:identity)
+
+      cleanup = fn v -> is_integer(v) and v > 0 end
+
+      sequence =
+        start_value
+        |> Enum.filter(cleanup)
+        |> Enum.map(&"START WITH #{&1}")
+        |> Kernel.++(
+          increment
+          |> Enum.filter(cleanup)
+          |> Enum.map(&"INCREMENT BY #{&1}")
+        )
+
+      case sequence do
+        [] -> [type_name, " GENERATED BY DEFAULT AS IDENTITY"]
+        _ ->  [type_name, " GENERATED BY DEFAULT AS IDENTITY(", Enum.join(sequence, " "), ") "]
+      end
+    end
+
     defp column_type(type, opts) do
       size      = Keyword.get(opts, :size)
       precision = Keyword.get(opts, :precision)
       scale     = Keyword.get(opts, :scale)
       type_name = ecto_to_db(type)
 
+      # handle when user has defined a column like
+      # add :id, :"bigint GENERATED BY DEFAULT AS IDENTITY"
+      type_name =
+        if type_name |> String.downcase() |> String.contains?(" generated") do
+          [actual_type, _] = String.split(type_name, " ", parts: 2)
+          actual_type
+        else
+          type_name
+        end
```

Comment:
Sorry, I mean to say to remove this code, but perhaps I don't understand exactly when it would be used. If we are skipping the generated part when doing `add :id, :"bigint GENERATED BY DEFAULT AS IDENTITY"`, then perhaps you should not pass the GENERATED value at all?

Reply:
My fault - I completely forgot that was still even there. It's been removed; I didn't intend to have that even in the original PR.